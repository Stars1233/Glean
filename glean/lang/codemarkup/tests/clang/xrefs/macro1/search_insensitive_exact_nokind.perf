{
  "@generated": null,
  "facts_searched": {
    "cxx1.DeclByName.5": 6,
    "cxx1.DeclarationNameSpan.5": 1,
    "cxx1.RecordDeclarationStruct.5": 1,
    "cxx1.RecordDefinition.5": 1,
    "pp1.Define.1": 1
  },
  "full_scans": [ ]
}