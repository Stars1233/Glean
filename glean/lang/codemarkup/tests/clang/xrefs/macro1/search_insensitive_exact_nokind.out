[
  "@generated",
  {
    "key": {
      "searchcase": 1,
      "name": "foo",
      "entity": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "Foo" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 25,
                      "columnBegin": 1,
                      "lineEnd": 29,
                      "columnEnd": 1
                    }
                  }
                },
                "bases": [ ],
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "foo" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "Foo" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "method": {
                            "isVirtual": false,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 26,
                            "columnBegin": 3,
                            "lineEnd": 26,
                            "columnEnd": 15
                          }
                        }
                      }
                    },
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "bar" } } },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "Foo" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "int" }
                                }
                              ]
                            }
                          },
                          "method": {
                            "isVirtual": false,
                            "isConst": false,
                            "isVolatile": false,
                            "refQualifier": 0
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 27,
                            "columnBegin": 3,
                            "lineEnd": 27,
                            "columnEnd": 18
                          }
                        }
                      }
                    },
                    {
                      "variable": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": "field" },
                              "scope": {
                                "recordWithAccess": {
                                  "record": {
                                    "key": {
                                      "name": { "key": "Foo" },
                                      "scope": { "global_": { } }
                                    }
                                  },
                                  "access": 0
                                }
                              }
                            }
                          },
                          "type": { "key": "int" },
                          "kind": { "field": { "mutable_": false } },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 28,
                            "columnBegin": 3,
                            "lineEnd": 28,
                            "columnEnd": 11
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      },
      "location": {
        "name": "Foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 25,
            "columnBegin": 1,
            "lineEnd": 29,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 453, "length": 3 }
        }
      },
      "kind": 24,
      "language": 0
    }
  },
  {
    "key": {
      "searchcase": 1,
      "name": "foo",
      "entity": {
        "pp": {
          "define": {
            "key": {
              "macro": { "key": "FOO" },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 9,
                "lineEnd": 9,
                "columnEnd": 11
              }
            }
          }
        }
      },
      "location": {
        "name": "FOO",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 9,
            "columnBegin": 9,
            "lineEnd": 9,
            "columnEnd": 11
          }
        }
      },
      "kind": 29,
      "language": 1
    }
  }
]